moduleframework/helpers/openshift_helper.py

```diff
@@ -54,8 +58,52 @@ def __init__(self):
         # application name is taken from docker.io/modularitycontainer/memcached
         self.app_name = self.container_name.split('/')[-1]
         self.app_ip = None
+        self.project_name = "%s-%s" % (''.join(random.choice(string.lowercase) for _ in range(3)), self.app_name)
         common.print_debug(self.icontainer, self.app_name)
 
+    def _get_openshift_ip_registry(self):
+        openshift_ip_register = None
+        docker_registry = self.runHost('oc get svc -n default -o json', ignore_status=True).stdout
+        docker_registry = self._convert_string_to_json(docker_registry)
+        for reg in docker_registry:
+            try:
+                name = reg.get("metadata").get("name")
+                if name == common.OPENSHIFT_DOCKER_REGISTER:
+                    openshift_ip_register = reg.get("spec").get("clusterIP")
+            except AttributeError:
+                pass
+
+        return openshift_ip_register
+
+    def _change_openshift_account(self, account="system:admin", password=None):
+        if password is None:
+            s = self.runHost("oc login -u %s" % account, verbose=common.is_not_silent())
+        else:
+            s = self.runHost("oc login -u %s -p %s" % (account, password), verbose=common.is_not_silent())
+
+
+    def _register_docker_to_openshift_register(self):
+        whoami = self.runHost("oc whoami -t", ignore_status=True, verbose=common.is_not_silent()).stdout
+        self._change_openshift_account()
+        self.runHost('docker pull %s' % self.container_name)
+        openshift_ip_register = self._get_openshift_ip_registry()
+        self._change_openshift_account(account=common.get_openshift_user(),
+                                password=common.get_openshift_passwd())
+
+        for i in range(0,5):
+            docker_login = self.runHost('docker login -u developer -p %s %s:5000' % (whoami, openshift_ip_register))
```

Comment:
"developer" is hardcoded - what if someone deploys openshift without that account?